tests/unit/test_pareto.py

```diff
@@ -111,3 +111,22 @@ def test_pareto_2d_bounds() -> None:
     npt.assert_allclose(
         pareto_2d.front, tf.constant([[0.1419, 0.9340], [0.1576, 0.7922], [0.4854, 0.0357]])
     )
+
+
+def test_hypervolume() -> None:
+    objectives = tf.constant(
+        [
+            [0.9575, 0.4218],
+            [0.9649, 0.9157],
+            [0.1576, 0.7922],
+            [0.9706, 0.9595],
+            [0.9572, 0.6557],
+            [0.4854, 0.0357],
+            [0.8003, 0.8491],
+            [0.1419, 0.9340],
+        ]
+    )
+
+    pareto_2d = Pareto(objectives)
+
+    npt.assert_allclose(pareto_2d.hypervolume_indicator(tf.constant([2.0, 2.0])), 3.3878, rtol=1e-2)
```

Comment:
how did you calculate this?

Reply:
I am using the same value as they use in https://github.com/GPflow/GPflowOpt/blob/3d86bcc000b0367f19e9f03f4458f5641e5dde60/testing/unit/test_pareto.py#L63

Reply:
do you know how they came across this value?

Reply:
the reason I ask is because it's difficult to tell by looking at this test whether it's right or not

Reply:
sorry for late response, I just confirmed it with Ivo. so in the gpflowopt they just try the algorithm (fully with acquisition function and BO steps) on some benchmark function. and see if everything works correctly and use both the pareto and the hypervolume of that pareto (3.3878) for the test. 
It seems strange way to do it actually.

Reply:
one of the option that we can use as Ivo suggest is drawing random points from upper right part of unit circle.  if the reference point is [1,1] the volumetric indicator should be close to 0.5*0.5 (the area of upper right square) - 1/4 of the circle area.

Reply:
would it not be possible to simply use some hand-chosen pareto points and work out the hypervolume by hand? To take a really simply case, a pareto front of ([-1, -0.5], [-0.5, -1]) with reference [0, 0] would give a hypervolume of 0.75 assuming I've understood correctly

Reply:
we'd also need tests for both an empty pareto front and for a single point. If the expected behaviour for an empty point is undefined, the method should error

Reply:
> 
> 
> would it not be possible to simply use some hand-chosen pareto points and work out the hypervolume by hand? To take a really simply case, a pareto front of ([-1, -0.5], [-0.5, -1]) with reference [0, 0] would give a hypervolume of 0.75 (do check that - I'm not certain I've got hypervolume yet). Probably worth using more interesting points and reference, but you get my drift

I don't have any interesting points in mind right now. but your sample indeed pretty easy to visualize and think of. I'll add it. If there are interesting points in the future, we can always add them as additional test points.

Reply:
> 
> 
> we'd also need tests for both an empty pareto front and for a single point. If the expected behaviour for an empty point is undefined, the method should error

okay

Reply:
how about I make some tests and send them your way? We should add the tests with the code

Reply:
okay